tests/drivers/spi/spi_loopback/overlay-rpi-pico-pio.conf

```diff
@@ -0,0 +1 @@
+CONFIG_SPI_ASYNC=n
```

Comment:
nit:

```
$ find ./tests/ -name '*.conf'  | xargs -i basename {} | grep -- '-' | wc -l
71
$ find ./tests/ -name '*.conf'  | xargs -i basename {} | grep -- '_' | wc -l
887
```

`_` is still the majority, so that's probably better.

The rest is LGTM.

Reply:
I noticed the rest of the config files in that folder have all been using `-`:
```
./overlay-sam-spi-dma.conf
./overlay-max32-spi-rtio.conf
./overlay-stm32-spi-dma-dt-nocache-mem.conf
./overlay-gd32-spi-interrupt.conf
./prj.conf
./overlay-max32-spi-dma.conf
./overlay-stm32-spi-16bits.conf
./overlay-stm32-spi-16bits-dma.conf
./overlay-stm32-spi-interrupt.conf
./overlay-rpi-pico-pio.conf
./overlay-stm32-spi-16bits-dma-dt-nocache-mem.conf
./overlay-stm32-spi-dma.conf
./overlay-gd32-spi-dma.conf
./overlay-mcux-dspi-dma.conf
```

If I have to rename, I'd better rename all of these. What do you think of it?

Reply:
Ah, I got it.
It seems the `-` is local custom of this test.
Let's keep current code.